tests/model_serving/model_server/components/kserve_dsc_deployment_mode/conftest.py

```diff
@@ -15,30 +15,58 @@
 from tests.model_serving.model_server.components.kserve_dsc_deployment_mode.utils import (
     patch_dsc_default_deployment_mode,
 )
+from utilities import constants
 from utilities.constants import ModelAndFormat
 from utilities.inference_utils import create_isvc
 
 
+@pytest.fixture(scope="class")
+def inferenceservice_config_cm(admin_client: DynamicClient) -> ConfigMap:
+    return ConfigMap(
+        client=admin_client,
+        name="inferenceservice-config",
+        namespace=py_config["applications_namespace"],
+    )
+
+
 @pytest.fixture(scope="class")
 def default_deployment_mode_in_dsc(
     request: FixtureRequest,
     dsc_resource: DataScienceCluster,
     inferenceservice_config_cm: ConfigMap,
 ) -> Generator[DataScienceCluster, Any, Any]:
-    yield from patch_dsc_default_deployment_mode(
+    patch_value = request.param["default-deployment-mode"]
```

Comment:
have an option to either pass `default-deployment-mode` or `raw-deployment-service-config`
Then:
```
if development_mode := request.param.get("default-deployment-mode"):
        spec_key = "defaultDeploymentMode"
        nested_path = ["defaultDeploymentMode"]
        config_key = "deploy"
        expected_value = development_mode
elif service_config := request.param.get("raw-deployment-service-config"):
       spec_key = "rawDeploymentServiceConfig"
        nested_path = ["serviceClusterIPNone"]
        config_key = "service"
        expected_value = service_config == constants.DscComponents.RawDeploymentServiceConfig.HEADLESS
else:
   raise ...
```